### PR TITLE
Expose action with entity

### PR DESCRIPTION
**Pull request by seadowg**
Work for getodk/collect#6031

This will allow a client to determine that an exposed entity is an update and handle edge cases around that (like the one described in the attached issue).

#### What has been done to verify that this works as intended?

Updated tests.

#### Why is this the best possible solution? Were any other approaches considered?

This is quite a simple change. I considered also removing the versioning logic from JavaRosa now that clients will have access to the action and theoretically could handle it for themselves. However, I wanted to keep the entity exposed as simple to deal with as possible - an `UPDATE` is communicating "update the entity with this id to this version with this label and properties".

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

Shouldn't change anything on client side as the change is purely additive.

### CODE REVIEW

**Review by lognaturel — APPROVED** [review]

File: src/test/java/org/javarosa/entities/EntityFormParserTest.java [hunk 1]
```diff
@@ -52,8 +52,8 @@ public void parseAction_findsCreateWithTrueString() throws XFormParser.ParseExce
         XFormParser parser = new XFormParser(new InputStreamReader(new ByteArrayInputStream(form.asXml().getBytes())));
         FormDef formDef = parser.parse(null);
 
-        EntityFormParser.EntityAction dataset = EntityFormParser.parseAction(EntityFormParser.getEntityElement(formDef.getMainInstance()));
-        assertThat(dataset, equalTo(EntityFormParser.EntityAction.CREATE));
+        EntityAction dataset = EntityFormParser.parseAction(EntityFormParser.getEntityElement(formDef.getMainInstance()));
```
Comment (lognaturel) [inline src/test/java/org/javarosa/entities/EntityFormParserTest.java:55]: Not: dataset seems like the wrong var name here. Action?

**Review by lognaturel — APPROVED** [review]
I’ll let you merge in case you want to update wording first!
